File: Iteration2/code/CSE3063GRP12/src/test/StudentModelTest.java
package test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotEquals;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

import core.models.abstracts.User;
import core.models.concretes.Student;

public class StudentModelTest {

    @Test
    public void testUserEqual() {
        Student student = new Student();
        student.setUserName("test");
        student.setPassword("test");
        Student student2 = new Student();
        student2.setUserName("test");
        student2.setPassword("test");

        assertEquals(student, student2);

    }

    @Test
    public void testUserNotEqual() {
        Student student = new Student();
        student.setUserName("test");
        student.setPassword("test");
        Student student2 = new Student();
        student2.setUserName("test1");
        student2.setPassword("test1");

        assertNotEquals(student, student2);

    }

    @Test
    public void testUserType() {
        User student = new Student();
        assertTrue(student instanceof User);
    }

}

File: Iteration3/code/CSE3063GRP12/src/StudentModelTest.py
import unittest
from core.models.concretes.Student import Student  # Assuming your Student class is in 'your_module'
from core.models.abstracts.User import User
from core.models.concretes.Advisor import Advisor

class StudentModelTest(unittest.TestCase):

    def test_user_equal(self):
        student = Student()
        student.userName = "test"
        student.password = "test"
        student2 = Student()
        student2.userName = "test"
        student2.password = "test"

        self.assertEqual(student, student2)

    def test_user_not_equal(self):
        student = Student()
        student.userName= "test"
        student.password = "test"
        student2 = Student()
        student2.userName= "test1"
        student2.password = "test1"

        self.assertNotEqual(student, student2)

    def test_user_type(self):
        student = Student()
        self.assertIsInstance(student, User)

if __name__ == '__main__':
    unittest.main()